platform/gce.go

```diff
@@ -44,10 +46,11 @@ type GCEOptions struct {
 }
 
 type gceCluster struct {
-	api      *compute.Service
-	sshAgent *network.SSHAgent
-	conf     *GCEOptions
-	machines map[string]*gceMachine
+	api        *compute.Service
+	sshAgent   *network.SSHAgent
+	conf       *GCEOptions
+	machines   map[string]*gceMachine
+	sync.Mutex // protects only against NewMachine being called in parallel
```

Comment:
Hmm, thats fair I will lock all access to the map while we are here. 

I wanted to convey that NewMachine() was the only part of the interface expected to work in parallel since we essentially are locking code by going back and making this so in the first place. Its generally not expected for the cluster to be something that thread safe overall. I guess user's of a cluster can still assume that the cluster is not a thread safe interface.